scenarios/remote_signals/src/main.rs

```diff
@@ -0,0 +1,238 @@
+use anyhow::Context;
+use holochain_types::prelude::*;
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+use remote_signal_integrity::TimedMessage;
+use std::collections::HashSet;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use trycp_wind_tunnel_runner::embed_conductor_config;
+use trycp_wind_tunnel_runner::prelude::*;
+
+embed_conductor_config!();
+
+#[derive(Debug)]
+pub struct ScenarioValues {
+    signal_interval: Duration,
+    response_timeout: Duration,
+    remote_signal_peers: Vec<AgentPubKey>,
+    pending_set: Arc<Mutex<HashSet<TimedMessage>>>,
+}
+
+fn env_dur(n: &'static str, d: u64) -> Duration {
+    match std::env::var(n) {
+        Ok(n) => Duration::from_millis(n.parse::<u64>().unwrap()),
+        _ => Duration::from_millis(d),
+    }
+}
+
+impl Default for ScenarioValues {
+    fn default() -> Self {
+        let signal_interval = env_dur("SIGNAL_INTERVAL_MS", 1_000);
+        let response_timeout = env_dur("RESPONSE_TIMEOUT_MS", 20_000);
+
+        Self {
+            signal_interval,
+            response_timeout,
+            remote_signal_peers: Vec::new(),
+            pending_set: Arc::new(Mutex::new(HashSet::new())),
+        }
+    }
+}
+
+impl UserValuesConstraint for ScenarioValues {}
+
+fn agent_setup(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    connect_trycp_client(ctx)?;
+    reset_trycp_remote(ctx)?;
+
+    let pending_set = ctx.get().scenario_values.pending_set.clone();
+
+    let client = ctx.get().trycp_client();
+    let agent_name = ctx.agent_name().to_string();
+    let reporter = ctx.runner_context().reporter();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            client
+                .configure_player(agent_name.clone(), conductor_config().to_string(), None)
+                .await?;
+
+            client.startup(agent_name.clone(), None).await?;
+
+            let _rcv_task = tokio::task::spawn(async move {
+                loop {
+                    match client.recv_signal().await {
+                        None => panic!("signal receiver ended"),
+                        Some(signal) => {
+                            let msg: Signal = ExternIO(signal.data).decode().unwrap();
```

Comment:
This could be an `expect()` call

```rust
let signal = client.recv_signal().await.expect("signal receiver ended");
let msg: Signal = ExternIO(signal.data).decode().unwrap();
// ...
```

Reply:
good call, updated: https://github.com/holochain/wind-tunnel/pull/93/commits/964b485cf80853023c24362506363693590aabfb